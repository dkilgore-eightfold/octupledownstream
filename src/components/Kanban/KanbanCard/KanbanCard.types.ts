import { ReactNode } from 'react';
import { KanbanLocale } from '../Kanban.types';
import { ButtonProps } from '../../Button';
import { CardProps } from '../../Card';
import { CheckboxProps } from '../../CheckBox';

export interface KanbanCardProps extends CardProps {
  /**
   * The card body content renderer.
   */
  body?: ReactNode;
  /**
   * The card checkbox props.
   */
  checkboxProps?: CheckboxProps;
  /**
   * Whether or not the card is checked.
   * @default false
   */
  checked?: boolean;
  /**
   * The card custom classes.
   */
  classNames?: string;
  /**
   * The collapse button aria label.
   */
  collapseButtonAriaLabel?: string;
  /**
   * The deselect card checkbox aria label.
   */
  deselectCardAriaLabel?: string;
  /**
   * Whether or not the card is expandable.
   * @default true
   */
  expandable?: boolean;
  /**
   * The expand button aria label.
   */
  expandButtonAriaLabel?: string;
  /**
   * The expand button props.
   */
  expandButtonProps?: ButtonProps;
  /**
   * The card expanded content renderer.
   */
  expandedContent?: ReactNode;
  /**
   * The card header content renderer.
   */
  header?: ReactNode;
  /**
   * Custom header classes.
   */
  headerClassNames?: string;
  /**
   * Whether or not the card is hoverable.
   * @default true
   */
  hoverable?: boolean;
  /**
   * The card hover content renderer.
   */
  hoverContent?: ReactNode;
  /**
   * The card id.
   */
  id?: string;
  /**
   * Whether or not the card is loading.
   */
  loading?: boolean;
  /**
   * Custom loading content of the card.
   */
  loadingContent?: ReactNode;
  /**
   * The kanban card locale.
   * @default 'enUS'
   */
  locale?: KanbanLocale;
  /**
   * Callback when the kanban card checkbox has changed.
   */
  onCheckboxChange?: (encId: string, checked: boolean) => void;
  /**
   * Callback when the kanban card loading has changed.
   * @default false
   */
  onLoadingChange?: (loading: boolean) => void;
  /**
   * The card is selectable and the checkbox is displayed.
   * @default false
   */
  selectable?: boolean;
  /**
   * The select card checkbox aria label.
   */
  selectCardAriaLabel?: string;
  /**
   * Callback when the card dimensions have changed.
   */
  updateDimension?: (async?: boolean, delay?: number) => void;
}
