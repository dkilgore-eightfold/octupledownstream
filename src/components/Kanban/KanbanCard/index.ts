export * from './KanbanCard';
export * from './KanbanCard.types';
