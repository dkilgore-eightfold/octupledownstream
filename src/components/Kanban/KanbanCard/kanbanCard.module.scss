.card-wrapper {
  .card-outer-wrapper {
    padding: $space-xxxs $space-m calc($space-xs - $space-xxxs);

    .card {
      background: var(--kanban-card-background-color);
      border: var(--kanban-card-border);
      border-radius: var(--kanban-card-border-radius);
      color: var(--kanban-card-text-color);
      font-family: var(--kanban-card-font-family);
      height: fit-content;
      transition: border-radius 150ms;
      width: 100%;

      &:hover {
        box-shadow: var(--kanban-card-hover-box-shadow);
      }
    }
  }
}

.checkbox {
  margin: $space-xxs;
}

.header {
  flex: 1;
  margin-left: -$space-xs;
  margin-top: $space-xxxs;
}

.expand-button {
  opacity: 0;
  transition: opacity $motion-duration-extra-fast $motion-easing-easeinout;

  &.visible {
    opacity: 1;
  }
}

.expanded-content {
  opacity: 0;
  pointer-events: none;

  &.expanded {
    opacity: 1;
    pointer-events: auto;
  }
}

.divider-expanded-card,
.divider-hover-actions {
  background: var(--kanban-card-divider-color);
  height: 1px;
  margin: $space-xs 0;
  width: 100%;
}
