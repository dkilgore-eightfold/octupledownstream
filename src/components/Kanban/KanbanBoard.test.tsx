import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import {
  KanbanBoard,
  KanbanBoardProps,
  KanbanItemWrapperProps,
  KanbanLaneGroupProps,
  KanbanLaneProps,
} from './';
import { KanbanCard } from './KanbanCard';
import { fireEvent, render, waitFor } from '@testing-library/react';
import mockDataSet from './mockdata.json';
import zhCN from './Locale/zh_CN';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

class ResizeObserver {
  observe() {
    // do nothing
  }
  unobserve() {
    // do nothing
  }
  disconnect() {
    // do nothing
  }
}

window.ResizeObserver = ResizeObserver;

interface MyConstants {
  board: MyKanbanBoard[];
}

interface MyKanbanDropzone {
  sub_lane: string;
}

interface MyKanbanBoard {
  display_name: string;
  lane_id: string;
  dropzones: MyKanbanDropzone[];
  expandable: boolean;
  'data-testid'?: string;
}

type MyKanbanState = {
  board: KanbanLaneGroupProps[];
  loading: Record<string, boolean>;
};

interface MyData {
  constants: MyConstants;
}

/**
 * Spoofs an API call to get data.
 * @returns MyData
 */
const getMockData = (): MyData => {
  const content: string = JSON.stringify(mockDataSet) || '{}';
  return JSON.parse(content);
};

const mockLoadedState = getMockData();

const myState: MyKanbanState = {
  board: mockLoadedState.constants?.board?.map((lane) => ({
    title: lane.display_name,
    laneId: lane.lane_id,
    'data-testid': lane['data-testid'],
    dropzones: lane.dropzones.map((dz) => ({
      id: dz.sub_lane,
      displayName: dz.sub_lane,
    })),
    expandable: lane.dropzones.length > 1,
  })),
  loading: {},
};

const mockProps: KanbanBoardProps = {
  classNames: 'kanban-board-test-class',
  id: 'kanbanBoardTestId',
  items: [0, 1, 2, 3, 4, 5, 6, 7].map((index: number) => ({
    id: `${'item-' + index}`,
    index,
    'data-testid': `${'item-' + index}`,
    dropzoneId: 'Lane 1',
    selected: false,
    render: () => (
      <KanbanCard
        selectable
        data-testid={`${'test-kanbancard-' + index}`}
        header={<div>{'Item ' + index} header</div>}
        body={<div>{'Item ' + index} body</div>}
        expandedContent={<div>{'Item ' + index} expanded content</div>}
        hoverContent={<div>{'Item ' + index} hover content</div>}
      />
    ),
  })),
  lanes: myState.board.map((lane: KanbanLaneGroupProps) => ({
    ...lane,
    laneStyles: {
      height: 'calc(100vh - 64px)', // Make sure to take into account header and/or footer height and insets when using viewport.
    },
    dropzones: lane.dropzones.map((dz) => ({
      ...dz,
    })),
    loadingProps: {
      loading: false,
      loadingItem: <KanbanCard loading />,
    },
  })),
  style: { background: 'red' },
  'data-testid': 'test-kanbanboard',
};

describe('KanbanBoard', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders KanbanBoard without crashing', () => {
    const { container, getByTestId } = render(
      <KanbanBoard {...mockProps} data-testid="test-kanbanboard" />
    );
    const kanbanBoard = getByTestId('test-kanbanboard');
    expect(() => container).not.toThrowError();
    expect(kanbanBoard).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Item drag and drop', () => {
    const { container, getByTestId } = render(<KanbanBoard {...mockProps} />);

    const draggableItem = getByTestId('test-kanbancard-0');
    const droppableZone = container.querySelector(
      '[data-rbd-droppable-id="LANE_1"]'
    );

    fireEvent.drag(draggableItem);
    fireEvent.drop(droppableZone);

    expect(droppableZone.innerHTML).toContain(draggableItem.innerHTML);
  });

  test('Renders KanbanBoard with custom classNames', () => {
    const { container } = render(<KanbanBoard {...mockProps} />);
    const kanbanBoard = container.querySelector('.kanban-board-test-class');

    expect(kanbanBoard).toBeTruthy();
  });

  test('Renders KanbanBoard with custom id', () => {
    const { container } = render(<KanbanBoard {...mockProps} />);
    const kanbanBoard = container.querySelector('#kanbanBoardTestId');

    expect(kanbanBoard).toBeTruthy();
  });

  test('Renders KanbanBoard with custom style', () => {
    const { getByTestId } = render(<KanbanBoard {...mockProps} />);
    const kanbanBoard = getByTestId('test-kanbanboard');

    expect(kanbanBoard.style.background).toBe('red');
  });

  test('Should apply laneGroupClassNames to KanbanLaneGroup', () => {
    const { container } = render(
      <KanbanBoard {...mockProps} laneGroupClassNames="test-lane-group" />
    );

    const laneGroupElement = container.querySelector('.test-lane-group');
    expect(laneGroupElement).toBeTruthy();
  });

  test('Should render with custom expandButtonAriaLabel', () => {
    const testExpandButtonAriaLabel = 'Custom expand';
    const { getByLabelText } = render(
      <KanbanBoard
        {...mockProps}
        expandButtonAriaLabel={testExpandButtonAriaLabel}
      />
    );

    const expandButton = getByLabelText(testExpandButtonAriaLabel);
    expect(expandButton).toBeTruthy();
  });

  test('Should render with custom collapseButtonAriaLabel', () => {
    const testExpandButtonAriaLabel = 'Custom expand';
    const testCollapseButtonAriaLabel = 'Custom collapse';
    const { getByLabelText } = render(
      <KanbanBoard
        {...mockProps}
        collapseButtonAriaLabel={testCollapseButtonAriaLabel}
        expandButtonAriaLabel={testExpandButtonAriaLabel}
      />
    );

    const expandButton = getByLabelText(testExpandButtonAriaLabel);
    fireEvent.click(expandButton);
    const collapseButton = getByLabelText(testCollapseButtonAriaLabel);
    expect(collapseButton).toBeTruthy();
  });

  test('Should render KanbanLaneLoader when loadingProps is provided', () => {
    const testLoadingProps = {
      lanesCount: 3,
      loading: true,
      loadingItem: <div data-testid="kanban-lane-loader">Loading...</div>,
    };

    const { getAllByTestId } = render(
      <KanbanBoard {...mockProps} loadingProps={testLoadingProps} />
    );

    const laneLoaders = getAllByTestId('kanban-lane-loader');
    expect(laneLoaders).toBeTruthy();
  });

  test('Should render KanbanLaneGroup when loadingProps is not provided', () => {
    const { container } = render(<KanbanBoard {...mockProps} />);

    const laneGroup = container.getElementsByClassName('lane-group')[0];
    expect(laneGroup).toBeTruthy();
  });

  test('Should render KanbanLaneGroup for each lane in the lanes prop', () => {
    const { getByTestId } = render(<KanbanBoard {...mockProps} />);

    const laneGroup1 = getByTestId('kanban-lane-1');
    const laneGroup2 = getByTestId('kanban-lane-2');

    expect(laneGroup1).toBeTruthy();
    expect(laneGroup2).toBeTruthy();
  });

  test('Should render with the provided locale', () => {
    const { getByLabelText } = render(
      <KanbanBoard {...mockProps} locale={zhCN} />
    );

    const expandButton = getByLabelText('扩大');
    expect(expandButton).toBeTruthy();
  });

  test('Renders lanes with correct titles', () => {
    const { getByTestId, getByText } = render(<KanbanBoard {...mockProps} />);

    mockProps.lanes.forEach(
      async (lane: KanbanLaneProps, index: number): Promise<void> => {
        await waitFor(() => getByText('Lane ' + index));
        const laneElement: HTMLElement = getByTestId(lane['data-testid']);
        expect(laneElement).toBeTruthy();
      }
    );
  });

  test('Renders items correctly', () => {
    const { getByTestId, getByText } = render(<KanbanBoard {...mockProps} />);

    mockProps.items.forEach(
      async (item: KanbanItemWrapperProps, index: number) => {
        await waitFor(() => getByText('Item ' + index));
        const itemElement: HTMLElement = getByTestId(item['data-testid']);
        expect(itemElement).toBeTruthy();
      }
    );
  });

  test('Toggles the expand button', () => {
    const { container, getByLabelText } = render(
      <KanbanBoard
        {...mockProps}
        collapseButtonAriaLabel="Collapse test"
        expandButtonAriaLabel="Expand test"
      />
    );

    const expandButton = getByLabelText('Expand test');

    expect(expandButton).toBeTruthy();

    fireEvent.click(expandButton);
    expect(container.querySelector('.expanded')).toBeTruthy();

    const collapseButton = getByLabelText('Collapse test');

    expect(collapseButton).toBeTruthy();

    fireEvent.click(collapseButton);
    expect(container.querySelector('.expanded')).toBeFalsy();
  });
});
