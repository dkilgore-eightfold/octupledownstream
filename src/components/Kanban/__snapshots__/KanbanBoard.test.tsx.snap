// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KanbanBoard Renders KanbanBoard without crashing 1`] = `
<div>
  <div
    class="stack kanban-board kanban-board-test-class horizontal undefined gap-m"
    data-testid="test-kanbanboard"
    id="kanbanBoardTestId"
    style="background: red;"
  >
    <div
      class="animated-wrapper"
      style="min-width: 400px;"
    >
      <div
        class="lane-group"
        style="height: calc(100vh - 64px);"
      >
        <div
          class="stack header full-width horizontal undefined"
          style="align-items: center; justify-content: space-between;"
        >
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          >
            <span>
              Lane 1
            </span>
            <span
              class="badge badge-medium"
            >
              8
            </span>
          </div>
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          />
        </div>
        <div
          class="lane-container"
          data-testid="kanban-lane-1"
          style="opacity: 0;"
        >
          <div
            class="lane"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="LANE_1"
          >
            <div
              style="overflow: visible; height: 0px; width: 0px;"
            >
              <div
                class="list"
                style="position: relative; width: 400px; overflow: auto; will-change: transform; direction: ltr;"
              >
                <div
                  style="height: 850px; width: 100%;"
                >
                  <div
                    style="position: absolute; left: 0px; top: 0px; height: 200px; width: 100%;"
                  >
                    <div
                      aria-describedby="rbd-hidden-text-0-hidden-text-0"
                      class="draggable-item-wrapper"
                      data-rbd-drag-handle-context-id="0"
                      data-rbd-drag-handle-draggable-id="item-0"
                      data-rbd-draggable-context-id="0"
                      data-rbd-draggable-id="item-0"
                      draggable="false"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="draggable-item"
                        data-testid="item-0"
                      >
                        <div
                          class="card-wrapper"
                        >
                          <div
                            class="card-outer-wrapper"
                          >
                            <div
                              class="card card card-bordered list card-small"
                              data-testid="test-kanbancard-0"
                              focus-visible-id="focus-visible-target"
                              id="hep7bo2"
                            >
                              <div
                                class="stack full-width vertical undefined gap-s"
                                dir="ltr"
                              >
                                <div
                                  class="stack full-width horizontal undefined gap-xs"
                                  dir="ltr"
                                  style="align-items: flex-start; justify-content: space-between;"
                                >
                                  <div
                                    class="selector selector-medium checkbox"
                                  >
                                    <input
                                      aria-disabled="false"
                                      aria-label="Select card"
                                      id="tg7w72m"
                                      readonly=""
                                      type="checkbox"
                                      value=""
                                    />
                                    <label
                                      class=""
                                      for="tg7w72m"
                                    >
                                      <span
                                        class="checkmark"
                                      />
                                      <span
                                        class="selector-label selector-label-end"
                                      />
                                    </label>
                                  </div>
                                  <div
                                    class="header"
                                  >
                                    <div>
                                      Item 0
                                       header
                                    </div>
                                  </div>
                                  <button
                                    aria-disabled="false"
                                    aria-label="Expand"
                                    aria-pressed="false"
                                    class="expand-button button button-system-ui button-medium round-shape icon-left"
                                  >
                                    <span
                                      aria-hidden="false"
                                      class="icon icon-wrapper"
                                      role="presentation"
                                    >
                                      <svg
                                        role="presentation"
                                        style="width: 20px; height: 20px;"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                                          style="fill: currentColor;"
                                        />
                                      </svg>
                                    </span>
                                  </button>
                                </div>
                                <div>
                                  Item 0
                                   body
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="expanded-content"
                                  >
                                    <div
                                      class="divider-expanded-card"
                                    />
                                    <div>
                                      <div>
                                        Item 0
                                         expanded content
                                      </div>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="stack full-width vertical undefined gap-m"
                                    dir="ltr"
                                    style="justify-content: flex-end;"
                                  >
                                    <div
                                      class="divider-hover-actions"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    style="position: absolute; left: 0px; top: 200px; height: 200px; width: 100%;"
                  >
                    <div
                      aria-describedby="rbd-hidden-text-0-hidden-text-0"
                      class="draggable-item-wrapper"
                      data-rbd-drag-handle-context-id="0"
                      data-rbd-drag-handle-draggable-id="item-1"
                      data-rbd-draggable-context-id="0"
                      data-rbd-draggable-id="item-1"
                      draggable="false"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="draggable-item"
                        data-testid="item-1"
                      >
                        <div
                          class="card-wrapper"
                        >
                          <div
                            class="card-outer-wrapper"
                          >
                            <div
                              class="card card card-bordered list card-small"
                              data-testid="test-kanbancard-1"
                              focus-visible-id="focus-visible-target"
                              id="7ediw4p"
                            >
                              <div
                                class="stack full-width vertical undefined gap-s"
                                dir="ltr"
                              >
                                <div
                                  class="stack full-width horizontal undefined gap-xs"
                                  dir="ltr"
                                  style="align-items: flex-start; justify-content: space-between;"
                                >
                                  <div
                                    class="selector selector-medium checkbox"
                                  >
                                    <input
                                      aria-disabled="false"
                                      aria-label="Select card"
                                      id="hyyjll5"
                                      readonly=""
                                      type="checkbox"
                                      value=""
                                    />
                                    <label
                                      class=""
                                      for="hyyjll5"
                                    >
                                      <span
                                        class="checkmark"
                                      />
                                      <span
                                        class="selector-label selector-label-end"
                                      />
                                    </label>
                                  </div>
                                  <div
                                    class="header"
                                  >
                                    <div>
                                      Item 1
                                       header
                                    </div>
                                  </div>
                                  <button
                                    aria-disabled="false"
                                    aria-label="Expand"
                                    aria-pressed="false"
                                    class="expand-button button button-system-ui button-medium round-shape icon-left"
                                  >
                                    <span
                                      aria-hidden="false"
                                      class="icon icon-wrapper"
                                      role="presentation"
                                    >
                                      <svg
                                        role="presentation"
                                        style="width: 20px; height: 20px;"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                                          style="fill: currentColor;"
                                        />
                                      </svg>
                                    </span>
                                  </button>
                                </div>
                                <div>
                                  Item 1
                                   body
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="expanded-content"
                                  >
                                    <div
                                      class="divider-expanded-card"
                                    />
                                    <div>
                                      <div>
                                        Item 1
                                         expanded content
                                      </div>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="stack full-width vertical undefined gap-m"
                                    dir="ltr"
                                    style="justify-content: flex-end;"
                                  >
                                    <div
                                      class="divider-hover-actions"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    style="position: absolute; left: 0px; top: 400px; height: 200px; width: 100%;"
                  >
                    <div
                      aria-describedby="rbd-hidden-text-0-hidden-text-0"
                      class="draggable-item-wrapper"
                      data-rbd-drag-handle-context-id="0"
                      data-rbd-drag-handle-draggable-id="item-2"
                      data-rbd-draggable-context-id="0"
                      data-rbd-draggable-id="item-2"
                      draggable="false"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        class="draggable-item"
                        data-testid="item-2"
                      >
                        <div
                          class="card-wrapper"
                        >
                          <div
                            class="card-outer-wrapper"
                          >
                            <div
                              class="card card card-bordered list card-small"
                              data-testid="test-kanbancard-2"
                              focus-visible-id="focus-visible-target"
                              id="2xq00fi"
                            >
                              <div
                                class="stack full-width vertical undefined gap-s"
                                dir="ltr"
                              >
                                <div
                                  class="stack full-width horizontal undefined gap-xs"
                                  dir="ltr"
                                  style="align-items: flex-start; justify-content: space-between;"
                                >
                                  <div
                                    class="selector selector-medium checkbox"
                                  >
                                    <input
                                      aria-disabled="false"
                                      aria-label="Select card"
                                      id="lj3pdnd"
                                      readonly=""
                                      type="checkbox"
                                      value=""
                                    />
                                    <label
                                      class=""
                                      for="lj3pdnd"
                                    >
                                      <span
                                        class="checkmark"
                                      />
                                      <span
                                        class="selector-label selector-label-end"
                                      />
                                    </label>
                                  </div>
                                  <div
                                    class="header"
                                  >
                                    <div>
                                      Item 2
                                       header
                                    </div>
                                  </div>
                                  <button
                                    aria-disabled="false"
                                    aria-label="Expand"
                                    aria-pressed="false"
                                    class="expand-button button button-system-ui button-medium round-shape icon-left"
                                  >
                                    <span
                                      aria-hidden="false"
                                      class="icon icon-wrapper"
                                      role="presentation"
                                    >
                                      <svg
                                        role="presentation"
                                        style="width: 20px; height: 20px;"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
                                          style="fill: currentColor;"
                                        />
                                      </svg>
                                    </span>
                                  </button>
                                </div>
                                <div>
                                  Item 2
                                   body
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="expanded-content"
                                  >
                                    <div
                                      class="divider-expanded-card"
                                    />
                                    <div>
                                      <div>
                                        Item 2
                                         expanded content
                                      </div>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  style="height: 0px; opacity: 0; margin-top: -8px;"
                                >
                                  <div
                                    class="stack full-width vertical undefined gap-m"
                                    dir="ltr"
                                    style="justify-content: flex-end;"
                                  >
                                    <div
                                      class="divider-hover-actions"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="dropzone-container"
          >
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 1"
            >
              <div
                class="dropzone-name"
              >
                Lane 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="animated-wrapper"
      style="min-width: 400px;"
    >
      <div
        class="lane-group"
        style="height: calc(100vh - 64px);"
      >
        <div
          class="stack header full-width horizontal undefined"
          style="align-items: center; justify-content: space-between;"
        >
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          >
            <span>
              Lane 2
            </span>
            <span
              class="badge badge-medium"
            >
              0
            </span>
          </div>
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          >
            <button
              aria-disabled="false"
              aria-label="Expand"
              aria-pressed="false"
              class="button button-system-ui button-medium round-shape icon-left"
            >
              <span
                aria-hidden="false"
                class="icon icon-wrapper"
                role="presentation"
              >
                <svg
                  role="presentation"
                  style="width: 20px; height: 20px;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 6H6V17H2V6M7 19H17V4H7V19M9 6H15V17H9V6M18 6H22V17H18V6Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="lane-container"
          data-testid="kanban-lane-2"
          style="opacity: 0;"
        >
          <div
            class="lane"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="LANE_2"
          >
            <div
              style="overflow: visible; height: 0px; width: 0px;"
            >
              <div
                class="list"
                style="position: relative; width: 400px; overflow: auto; will-change: transform; direction: ltr;"
              >
                <div
                  style="height: 0px; width: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="dropzone-container"
          >
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 2 Sub Lane 1"
            >
              <div
                class="dropzone-name"
              >
                Lane 2 Sub Lane 1
              </div>
            </div>
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 2 Sub Lane 2"
            >
              <div
                class="dropzone-name"
              >
                Lane 2 Sub Lane 2
              </div>
            </div>
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 2 Sub Lane 3"
            >
              <div
                class="dropzone-name"
              >
                Lane 2 Sub Lane 3
              </div>
            </div>
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 2 Sub Lane 4"
            >
              <div
                class="dropzone-name"
              >
                Lane 2 Sub Lane 4
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="animated-wrapper"
      style="min-width: 400px;"
    >
      <div
        class="lane-group"
        style="height: calc(100vh - 64px);"
      >
        <div
          class="stack header full-width horizontal undefined"
          style="align-items: center; justify-content: space-between;"
        >
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          >
            <span>
              Lane 3
            </span>
            <span
              class="badge badge-medium"
            >
              0
            </span>
          </div>
          <div
            class="stack horizontal undefined gap-xs"
            style="align-items: center;"
          />
        </div>
        <div
          class="lane-container"
          data-testid="kanban-lane-3"
          style="opacity: 0;"
        >
          <div
            class="lane"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="LANE_3"
          >
            <div
              style="overflow: visible; height: 0px; width: 0px;"
            >
              <div
                class="list"
                style="position: relative; width: 400px; overflow: auto; will-change: transform; direction: ltr;"
              >
                <div
                  style="height: 0px; width: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="dropzone-container"
          >
            <div
              class="dropzone"
              data-rbd-droppable-context-id="0"
              data-rbd-droppable-id="Lane 3"
            >
              <div
                class="dropzone-name"
              >
                Lane 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
